### PR TITLE
Update Client to handle 400 responses from the controller

### LINKED ISSUE #8: flynn create without an app name causes unexpected 500
```
$ flynn create
POST http://i69.flynnhub.com:443/apps: controller: unexpected status 500
```

Running `flynn create demo-app` works just fine though. To be honest, I'm not sure where this issue originates, but the error mentioned the controller, so I figured this would be a good place to start.

### PR DESCRIPTION
**Pull request by lmars**
Fixes #8 by returning a more informative error:

```
$ flynn create
controller error: name must not be blank

$ flynn create "&"
controller error: name is invalid
```

`ControllerError` is pretty similar to `ValidationError`, but I wasn't sure the best way to handle that duplication.

### CODE REVIEW

**Comment by titanous** [comment]
You can just move `ValidationError` into the `types` package, it's there to handle this exact situation (look for how we reference `ct.App`, `ct.Release`, etc in flynn-cli)

**Comment by lmars** [comment]
@titanous ah ok: 0718828

File: client/client.go [hunk 1]
```diff
@@ -124,6 +124,13 @@ func (c *Client) rawReq(method, path string, contentType string, in, out interfa
 		res.Body.Close()
 		return res, ErrNotFound
 	}
+	if res.StatusCode == 400 {
+		body := ct.ValidationError{}
```
Comment (titanous) [inline client/client.go]:
``` go
var body ct.ValidationError
```

File: client/client.go [hunk 2]
```diff
@@ -124,6 +124,13 @@ func (c *Client) rawReq(method, path string, contentType string, in, out interfa
...
+		if err = json.NewDecoder(res.Body).Decode(&body); err != nil {
```
Comment (titanous) [inline client/client.go:130]: Add a `defer res.Body.Close()` above this line.

File: client/client.go (hunk 1 above)
Comment (lmars) [inline client/client.go]: updated: 83b6d07

**Pull request merged**